### PR TITLE
Add Copy button for `GoalInfoScreen` notes card

### LINKED ISSUE #94: [Feature Request] Copy text in goal notes with long press
### Checklist

- [X] This issue contains only one feature request. I will open one issue for every feature I want to request.
- [X] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo.
- [X] This feature is merely an UI/UX update.
- [X] This feature is not going to conflict with many of the existing features/options.

### Is your feature request related to a problem? Please describe.

I would like to keep URL or other types of information in one place where I also track progress on the goal. Currently the text that gets saved can be accessed again by editing the goal again. Would like to have an option to quickly copy the text from the notes field.

### Describe the solution you'd like

Copying to clipboard after long pressing the text on notes (after expanding) or having a separate button that copies the note.

### How will you/everyone benefit from this feature?

Less steps in copying the text (URL or search keywords) into clipboard.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mrghosti3**
Adds a button for copying text from Note card in goal info screen.

Closes #94

### CODE REVIEW

**Comment by mrghosti3** [comment]
May need extra edits for translations but otherwise think the main functionality is complete.

File: app/src/main/res/values-es/strings.xml [hunk 1]
```diff
@@ -6,6 +6,8 @@
     <string name="cancel">Cancelar</string>
     <string name="ok">Aceptar</string>
     <string name="unknown_error">¡Vaya! Algo salió mal.</string>
+    <string name="copy">Copy</string>
```
Comment (starry-shivam) [inline app/src/main/res/values-es/strings.xml:9]: Please move strings related to copy, such as button title, toast message, and content description, to the [goal info screen block](https://github.com/Pool-Of-Tears/GreenStash/blob/823c0e1bf183e2e3809cf17e7b056cfb87b8cca5/app/src/main/res/values/strings.xml#L81) inside `strings.xml`. Additionally, as a suggestion, if possible, you can also add translations for these strings. As they're very small (ranging from one to a few words), it shouldn't be a problem. You can use ChatGPT's help for translation.

**Review by starry-shivam — CHANGES_REQUESTED** [review]
Thanks for your contribution! Left a few comments.

File: app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt [hunk 2]
```diff
@@ -164,5 +180,19 @@ fun ExpandableTextCard(
             overflow = TextOverflow.Ellipsis,
             modifier = Modifier.padding(start = 12.dp, end = 12.dp)
         )
+        if (showCopyButton) {
+            FilledTonalButton(onClick = {
+                clipboardManager.setText(AnnotatedString(description))
+                if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.S_V2) {
+                    Toast.makeText(context, R.string.copy_alert, Toast.LENGTH_SHORT).show()
```
[the remark this answers is not in the document]
Comment (mrghosti3) [reply app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt]: Fixed with [d121651](https://github.com/Pool-Of-Tears/GreenStash/pull/95/commits/d12165117a5f55ce3511b6b0e0b765e139b83731)

File: app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt [hunk 3]
```diff
@@ -164,5 +180,19 @@ fun ExpandableTextCard(
...
+                }
+            }) {
+                Row {
+                    Icon(Icons.Filled.ContentCopy, contentDescription = "Copy Notes" )
```
[the remark this answers is not in the document]
Comment (mrghosti3) [reply app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt]: Fixed with [42ed507](https://github.com/Pool-Of-Tears/GreenStash/pull/95/commits/42ed50700bd89d421b0cdc8cda02203326502405)

File: app/src/main/res/values-es/strings.xml (hunk 1 above)
Comment (mrghosti3) [reply app/src/main/res/values-es/strings.xml]: Strings moved and renamed to have a matching prefix in [42ed507](https://github.com/Pool-Of-Tears/GreenStash/pull/95/commits/42ed50700bd89d421b0cdc8cda02203326502405)

As for translations, I will look into them slightly later.

File: app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (starry-shivam) [reply app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt]: Sorry, but I couldn't find the icon size being added in the commit you mentioned.

File: app/src/main/res/values-es/strings.xml (hunk 1 above)
Comment (starry-shivam) [reply app/src/main/res/values-es/strings.xml]: Thanks! Translation was just a suggestion and if you want, you can also skip it, I'll add them later :)

File: app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrghosti3) [reply app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt]: Ah, I just changed the description to be set from String resource. Missed the Size modifier. Will fix that soon.

[the remark this answers is not in the document]
Comment (mrghosti3) [reply app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt]: Oki, the newest commit should add the size modifier for the Icon

File: app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt [hunk 4]
```diff
@@ -164,5 +180,19 @@ fun ExpandableTextCard(
             overflow = TextOverflow.Ellipsis,
             modifier = Modifier.padding(start = 12.dp, end = 12.dp)
         )
+        if (showCopyButton) {
+            FilledTonalButton(onClick = {
```
[the remark this answers is not in the document]
Comment (mrghosti3) [reply app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt]: Oki, added the padding with the newest commit

File: app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt [hunk 5]
```diff
@@ -164,5 +183,27 @@ fun ExpandableTextCard(
...
             modifier = Modifier.padding(start = 12.dp, end = 12.dp)
         )
+        if (showCopyButton) {
+            FilledTonalButton(
+                onClick = {
+                    clipboardManager.setText(AnnotatedString(description))
+                    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.S_V2) {
+                        getString(context, R.string.info_copy_alert)
+                            .toToast(context, Toast.LENGTH_SHORT)
+                    }
```
Comment (starry-shivam) [inline app/src/main/java/com/starry/greenstash/ui/common/ExpandableCard.kt:193]: You don't need to pass `Toast.LENGTH_SHORT` as its already the default value.

File: app/src/main/res/values-es/strings.xml [hunk 6]
```diff
@@ -92,6 +92,9 @@
     <string name="info_goal_no_transactions">Sin transacciones aún.</string>
     <string name="info_edit_transaction_button">Actualizar Transacción</string>
     <string name="info_transaction_onboarding_tip">Consejo: Desliza las transacciones hacia la izquierda o hacia la derecha para editarlas o eliminarlas.</string>
+    <string name="info_copy_button">Copy</string>
+    <string name="info_copy_icon_description">Copy Notes</string>
+    <string name="info_copy_alert">Copied</string>
```
Comment (starry-shivam) [inline app/src/main/res/values-es/strings.xml:97]: I think something like "Copied Successfully!" would look much better. What do you think?

**Review by starry-shivam — CHANGES_REQUESTED** [review]
Thanks for addressing the review, just a couple more nitpicks.

File: app/src/main/res/values-es/strings.xml (hunk 6 above)
Comment (mrghosti3) [reply app/src/main/res/values-es/strings.xml:97]: Yeah, sounds good to me, gonna add that change in the next commits

File: app/src/main/res/values-es/strings.xml (hunk 1 above)
Comment (mrghosti3) [reply app/src/main/res/values-es/strings.xml]: Then lets leave the translations as is for now. Can work on them more with later PRs.

**Comment by starry-shivam** [comment]
Alright! Things look pretty good to me now. Please clarify what you want to do about translations.

**Comment by starry-shivam** [comment]
Nvm, Got your response in the above comment.

**Review by starry-shivam — APPROVED** [review]

**Comment by starry-shivam** [comment]
Merged! Thanks again for contributing 🎉

**Comment by mrghosti3** [comment]
Thanks for accepting my contribution :partying_face:

**Pull request merged**